### ISSUE TITLE
false negative - Type int for $<tmp coalesce var>336 is never null

### ISSUE THREAD

**Issue by maciejlebkowski** [vimeo/psalm#10491]
https://psalm.dev/r/230b2e4570

```php
<?php

final class Container {
	public function __construct(public readonly Estimate $estimate) {}
}

final class Estimate {
	public function __construct(public readonly int $value) {}
	public function value() : int { return $this->value; }
}

function foo(?Container $input) : void {
	print $input?->estimate->value ?? 0;
	print $input?->estimate->value() ?? 0; // line 14
}
```

> Psalm output (using commit [a75d26a](https://github.com/vimeo/psalm/commit/a75d26a)): 
> ERROR: [RedundantCondition](https://psalm.dev/122) - 14:8 - Type int for $<tmp coalesce var>336 is never null
> ERROR: [TypeDoesNotContainNull](https://psalm.dev/090) - 14:38 - Cannot resolve types for $<tmp coalesce var>336 - int does not contain null

Using a null-coalescing operator before a nested method call causes psalm to produce a false negative, claiming that the expression cannot be null. Same structure with property access instead of a method call works correctly, as observed.
